src/Container/ReadOnlyContainer.php

```diff
@@ -41,18 +41,18 @@ class ReadOnlyContainer implements ContainerInterface
      *
      * @param array<string, callable(ContainerInterface $container):mixed> $services
      * @param array<string, bool> $factoryIds
-     * @param array<string, array<callable(mixed, ContainerInterface $container):mixed>> $extensions
+     * @param ServiceExtensions|array $extensions
      * @param ContainerInterface[] $containers
      */
     public function __construct(
         array $services,
         array $factoryIds,
-        array $extensions,
+        $extensions,
```

Comment:
We might should consider raising the PHP version (currently min is 7.2) to make use of Union Types. 🤔

Reply:
We can not, if we want to use modularity in internal projects (products) that don't have the luxury to require PHP 8+

Reply:
True. MLP has still 7.4 | 8.0 declared. Maybe we can at least increase to 7.4 at some point. 🤔

Reply:
7.4 should be fine, yes.